### DISCUSSION TITLE
Activation Functions, or the Lack of Them, under Computer Vision.

### DISCUSSION THREAD

**Discussion by WALTEROBILA** [mrdbourke/pytorch-deep-learning#818] [Q&A]
When we were doing classification, we would use activation functions in the process of converting the predictions from logits, to probabilities, then finally to labels. For the "03_PyTorch_Computer_Vision" module, we only used argmax to convert the logits to labels. Any clarification on why this is the case?

**Comment by LuluW8071** [comment]
> When we were doing classification, we would use activation functions in the process of converting the predictions from logits, to probabilities, then finally to labels. For the "03_PyTorch_Computer_Vision" module, we only used argmax to convert the logits to labels. Any clarification on why this is the case?

Process of converting the _predictions_ from _logits_, to _probabilities_ is done for **Binary Classification**  mainly for `loss_fn = nn.BCEWithLogitsLoss` to find probs distribution over classes. We dont need such conversion process in Multiclass classification in "**03_PyTorch_Computer_Vision**" where there were _10 labels_

## Similar Discussion in here #314

**Reply by WALTEROBILA** [reply]
I'll have to differ with this. For the multiclass classification problem under **Putting it all together** timestamp **13:42:10**, we used the `torch.softmax` activation to convert the logits into probabilities.

**Reply by LuluW8071** [reply]
Oh thats because we needed to find probability distribution of colored dots.

In the "03_PyTorch_Computer_Vision", the way model is used requires only the prediction of the class with the highest probability,  not the full probability distribution over all classes. In such cases, applying the _softmax_ and then taking the _argmax_  might not provide any additional benefits over directly taking the  _argmax_ of the logits.

**Reply by WALTEROBILA** [reply]
This explanation clears it up. Thank you very much.
